repairnator/src/main/java/fr/inria/spirals/repairnator/Launcher.java

```diff
@@ -233,33 +245,40 @@ private void run(String[] args) throws JSAPException, IOException {
     }
 
     private void mainProcess() throws IOException {
-        String input = this.arguments.getString("input");
-        String workspace = arguments.getString("workspace");
-        int lookupDays = arguments.getInt("lookup");
-        String output = arguments.getString("output");
-        boolean debug = arguments.getBoolean("debug");
-        RepairMode mode = RepairMode.valueOf(arguments.getString("mode").toUpperCase());
-        int steps = Integer.parseInt(arguments.getString("steps"));
-        boolean clean = arguments.getBoolean("clean");
-        boolean push = arguments.getBoolean("push");
+        input = this.arguments.getString("input");
+        workspace = arguments.getString("workspace");
+        lookupDays = arguments.getInt("lookup");
+        output = arguments.getString("output");
+        debug = arguments.getBoolean("debug");
+        mode = RepairMode.valueOf(arguments.getString("mode").toUpperCase());
+        steps = Integer.parseInt(arguments.getString("steps"));
+        clean = arguments.getBoolean("clean");
+        push = arguments.getBoolean("push");
+        solverPath = arguments.getString("z3Path");
 
         if (debug) {
             setLevel(Level.DEBUG);
         }
+        
         JsonSerializer jsonSerializer = new JsonSerializer(output, mode);
         CSVSerializer csvSerializer = new CSVSerializer(output);
-        GoogleSpreadSheetInspectorSerializer googleSpreadsheetSerializer = new GoogleSpreadSheetInspectorSerializer();
+        GoogleSpreadSheetInspectorSerializer googleSpreadSheetInspectorSerializer = new GoogleSpreadSheetInspectorSerializer();
+
         GoogleSpreadSheetInspectorTimeSerializer googleSpreadSheetInspectorTimeSerializer = new GoogleSpreadSheetInspectorTimeSerializer();
 
         this.serializers.add(jsonSerializer);
         this.serializers.add(csvSerializer);
         this.serializers.add(googleSpreadSheetInspectorTimeSerializer);
-
-        if (push) {
-            this.serializers.add(googleSpreadsheetSerializer);
+        
+        if (push || mode == RepairMode.FORBEARS) {
+            this.serializers.add(googleSpreadSheetInspectorSerializer);
         }
 
-        Launcher.LOGGER.debug("Start to scan projects in travis for failing builds...");
+        if (mode != RepairMode.FORBEARS) {
+        	Launcher.LOGGER.debug("Start to scan projects in travis for failing builds...");
+        } else {
+        	Launcher.LOGGER.debug("Start to scan projects in travis for passing builds...");
+        }
```

Comment:
This condition is unnecessary: we can use a global log message like "Start to scan projects in travis"